packages/esm-patient-immunizations-app/src/immunizations/immunizations-form.component.tsx

```diff
@@ -119,14 +149,27 @@ const ImmunizationsForm: React.FC<DefaultWorkspaceProps> = ({ patientUuid, close
     updateSingle('currentDose', currentDose);
   };
 
+  if (isLoading || vaccineState === undefined)
+    return <span className={styles.vaccineLoader}>{`${t('loadingVaccine', 'Loading Vaccine')} ...`} </span>;
+
   return (
-    <Form className={styles.form} onSubmit={handleFormSubmit} data-testid="immunization-form">
-      <div>
-        <h4 className={styles.immunizationSequenceSelect}>
-          {`${t('vaccine', 'Vaccine')} : ${formState?.vaccineName}`}{' '}
-        </h4>
+    <Form className={styles.form} onSubmit={handleFormSubmit}>
+      <Stack gap={4}>
+        {vaccines?.length ? (
+          <section className={styles.immunizationSequenceSelect}>
+            <ComboBox
+              titleText="Vaccines"
```

Comment:
Any user-visible text should be localisable, e.g.,

```suggestion
              titleText={t('vaccineComboBoxTitle', 'Vaccines')}
```